packages/cli/src/environments.ee/source-control/source-control-status.service.ee.ts

```diff
@@ -606,4 +622,160 @@ export class SourceControlStatusService {
+		);
+
+		const projectsModifiedInEither: ExportableProjectWithFileName[] = [];
+
+		projectsLocal.forEach((localProject) => {
+			const remoteProjectWithSameId = projectsRemote.find(
+				(remoteProject) => remoteProject.id === localProject.id,
+			);
+
+			if (!remoteProjectWithSameId) {
+				return;
+			}
+
+			if (this.isProjectModified(localProject, remoteProjectWithSameId)) {
+				let name =
+					(options?.preferLocalVersion ? localProject?.name : remoteProjectWithSameId?.name) ??
+					'Project';
+
+				if (
+					localProject.name &&
+					remoteProjectWithSameId?.name &&
+					localProject.name !== remoteProjectWithSameId.name
+				) {
+					name = options?.preferLocalVersion
+						? `${localProject.name} (Remote: ${remoteProjectWithSameId.name})`
+						: `${remoteProjectWithSameId.name} (Local: ${localProject.name})`;
+				}
+
+				projectsModifiedInEither.push({
+					...localProject,
+					name,
+					description: options.preferLocalVersion
+						? localProject.description
+						: remoteProjectWithSameId.description,
+					icon: options.preferLocalVersion ? localProject.icon : remoteProjectWithSameId.icon,
+				});
+			}
+		});
+
+		const mapExportableProjectWithFileNameToSourceControlledFile = ({
+			project,
+			status,
+			conflict,
+		}: {
+			project: ExportableProjectWithFileName;
+			status: SourceControlledFile['status'];
+			conflict: boolean;
+		}): SourceControlledFile => {
+			return {
+				id: project.id,
+				name: project.name ?? 'Project',
+				type: 'project',
+				status,
+				location: options.direction === 'push' ? 'local' : 'remote',
+				conflict,
+				file: project.filename,
+				updatedAt: new Date().toISOString(),
+				owner: {
+					type: project.owner.type,
+					projectId: project.owner.teamId,
+					projectName: project.owner.teamName,
+				},
+			};
+		};
+
+		projectsMissingInLocal.forEach((item) => {
+			sourceControlledFiles.push(
+				mapExportableProjectWithFileNameToSourceControlledFile({
+					project: item,
+					status: options.direction === 'push' ? 'deleted' : 'created',
+					conflict: false,
+				}),
+			);
+		});
+
+		projectsMissingInRemote.forEach((item) => {
+			sourceControlledFiles.push(
+				mapExportableProjectWithFileNameToSourceControlledFile({
+					project: item,
+					status: options.direction === 'push' ? 'created' : 'deleted',
+					conflict: options.direction === 'push' ? false : true,
+				}),
+			);
+		});
+
+		projectsModifiedInEither.forEach((item) => {
+			sourceControlledFiles.push(
+				mapExportableProjectWithFileNameToSourceControlledFile({
+					project: item,
+					status: 'modified',
+					conflict: true,
+				}),
+			);
+		});
+
+		return {
+			projectsRemote,
+			projectsLocal,
+			projectsMissingInLocal,
+			projectsMissingInRemote,
+			projectsModifiedInEither,
+		};
+	}
+
+	private isProjectModified(
+		local: ExportableProjectWithFileName,
+		remote: ExportableProjectWithFileName,
+	): boolean {
+		const isIconModified =
```

Comment:
nit: I bit verbose, no ?
I think this does the same thing with less cognitive load:
```
const isIconModified =
  (!!remote.icon !== !!local.icon) ||
  (remote.icon?.type !== local.icon?.type || remote.icon?.value !== local.icon?.value);
```